mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2644,7 +2738,11 @@ static LogicalResult populateHostHarnessLogic(
       } else if ((genValidation == "clone") || elemType.isInteger(8) ||
                  elemType.isInteger(32)) {
         valElemType = elemType;
+      } else if (!gpuValidation && isSmallFloatIn &&
```

Comment:
Not GPU validation, small floats, and we're generating a kernel:  use the kernel's types for the validator.